app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java

```diff
@@ -247,5 +274,35 @@ Object getItem(int position) {
         return items.get(position);
     }
 
+
+    private int getItemCount(boolean excludeFooter) {
+        if (excludeFooter) return getItemCount();
```

Comment:
this logic seems backwards. If `excludeFooter` should return the whole count minus 1 (the footer).

Is there any guarantee that there's only 1 footer?  If not, it might be best to write a function to return the number of footers.